### PR TITLE
Apply some of the changes I've made to the dlvm branch, simplify some…

### PR DESCRIPTION
**Pull request by vimota**
… installations.

Have to see if it still builds since this branch isn't based on DLVM.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -116,11 +100,10 @@ RUN pip install matplotlib && \
     # ...as is psycopg2
     apt-get install -y libpq-dev && \
     pip install ibis-framework && \
-    # Cartopy plus dependencies
-    yes | conda install proj4 && \
-    pip install packaging && \
```
Comment (rosbo) [inline Dockerfile]: Are `proj4` and `packaging` going to get installed as dependencies of `Shapely` and `cartopy`?

File: Dockerfile [hunk 2]
```diff
@@ -69,10 +59,7 @@ RUN apt-get install -y libfreetype6-dev && \
     pip install nolearn && \
     pip install Theano && \
     pip install pybrain && \
-    # Base ATLAS
-    apt-get install -y libatlas-base-dev && \
-    cd /usr/local/src && git clone --depth 1 https://github.com/ztane/python-Levenshtein && \
-    cd python-Levenshtein && python setup.py install && \
+    pip install python-Levenshtein && \
```
Comment (rosbo) [inline Dockerfile]: adding a smoke test for this package to ensure there is no regresssion would be helpful. It should also be pretty straightforward given you can write a test asserting on the levenshtein distance between two words is equal to some expected distance.

File: Dockerfile (hunk 1 above)
Comment (vimota) [reply Dockerfile]: Proj4 definitely is (https://scitools.org.uk/cartopy/docs/latest/installing.html#installing) and interestingly enough so is Shapely - so I assumed from the comment all those  things we were installing were just requirements for Cartopy.

Looking at packaging (https://github.com/pypa/packaging) it doesn't seem like it's something that a user requested and more like it was installed as a dependency - do you have any background on it? (doing git blame didn't help)

Comment (rosbo) [reply Dockerfile]: I moved the conda commands to the top as recommended when using pip alongside conda. So far, it seems like it has solved our flakiness problem: https://github.com/Kaggle/docker-python/pull/760

Once my PR is in, you can merge master and keep your change for `Levenshtein` and `imagemagick` (don't forget to install it at the top along the other conda commands).

Thank you

**Comment by vimota** [comment]
Added a test, ran it locally and it passed!

**Pull request merged**